### ISSUE TITLE
Move visualizers out of JSON

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#1129]
In the attempt to remove `from_dict` (see #652, #720), it seems that the visualizers use JSON via transpiler (called with `format='json'`). We are moving to a transpiler that returns a DAG (and a property set). Visualizers should move out of using JSON. I guess, they should use DAG. Probably, that's a big refactor...

**Comment by mtreinish** [comment]
I'm not super familiar with the DAG class but I don't think it will be too bad to convert it. @1ucian0 you're familiar with the text drawer (I assume it works the same way), but the latex drawer and matplotlib only uses the json output to get a list of operations (with arguments and all the details) and then loops over that to generate the diagram. It also uses the headers to build the mapping of labels to bits. We will just need to be able to get is this iterator of operations from the DAG and the mapping. So as long as there is something we can use (or build) to get that information from the DAG it shouldn't be too hard.

But that being said, I really just want to build a method to extract all that information from the raw QuantumCircuit objects. There shouldn't be any reason we need to call transpile on the circuits to visualize it. I think it's just there to be able to pass the basis gates in as an argument and have transpile adjust the circuit. But, I feel this is out of scope for the circuit visualizers and we should deprecate and remove it anyway.

**Comment by ajavadia** [comment]
I agree with the premise of this PR. The transpiler format should only work on DAG. 'json' format of the transpiler should be fully removed. Each visualizere should work on a DAG.

The circuit visualizer should work on DAG, but not use the transpiler. That means remove 'basis' argument from the visualizer.

**Comment by 1ucian0** [comment]
Maybe every visualizar should have the same input. @mtreinish what do you think about using this common function and its output instead of the JSON?
```
def get_instructions(dag):
    ops = []
    for node_no in dag.nodes_in_topological_order:
        node = dag.multi_graph.node[node_no]
        if node['type'] == 'op':
            ops.append(node)
    return dag.qregs, dag.cregs, ops
```

**Comment by 1ucian0** [comment]
Other option is to have `qregs` and `cregs` in the right order and we include the `reversebits` logic there (def get_instructions(dag, reversebits)`)

**Comment by mtreinish** [comment]
@1ucian0 yeah using a common function (which is what the `qiskit.tools.visualization._utils` module is for) is exactly what I was advocating for. So instead of calling transpile we just run `DAGCircuit.fromQuantumCircuit()` and pass the DAG to that function. We'll just need to deprecate the basis kwarg on circuit_drawer() since we won't be able to do that with this approach (which I think is something we should do anyway)

**Comment by ajavadia** [comment]
Yes the basis kwarg to circuit_drawer() should be deprecated.

**Comment by 1ucian0** [comment]
So, do we agree with `def get_instructions(dag, reversebits) -> (dag.qregs, dag.cregs, ops)`?

**Comment by 1ucian0** [comment]
Okay... here is the plan:
1. We create `get_instructions` as a utility (PR #1187)
2. We create individual issues for the visualizers to move out of running the transpiler and use the new utility.
3. Once they are old moved, we removed the `format='json'` from the transpiler.

**Comment by 1ucian0** [comment]
PR #1187 merged. Converting this issue in an epic to track the spin-offs bugs.

**Comment by jaygambetta** [comment]
I did not comment on this before (did not see) but as I said on your pr I really don't see the point in making a dag when drawing a circuit. I would like the drawing module to only know about the circuit (even if the internals of the circuit become the dag) the object that we should send to this is the circuit not the dag. It is just a method for taking a circuit and returning a list of operations. 

Thanks

**Comment by 1ucian0** [comment]
mpl: #1272 fixed via #1278 
  latex_source: #1273 fixed via #1276 
  text: #1274 fixed via #1280
